### PR TITLE
Add tac utility

### PR DESCRIPTION
**Pull request by syrmel**
Following up on the work from PR# 84 and following the principle "Make it work, first, then optimize.", this version loads the entire file into memory and has potential to be improved.

I am experimenting with auto-generating boilerplate code (in the *.autogen.v files) from a simple JSON config file that specifies the utility name, description, and settings to ensure as much consistency as possible across utils.

### CODE REVIEW

File: src/uniq/uniq_test.v [hunk 1]
```diff
@@ -21,8 +21,8 @@ const posix_test_data = [
 	'#06 foo0 bar0 foo1 bar1',
 	'#07 bar0 foo1 bar1 foo0',
 ]
-const posix_test_path_newline = 'posix_nl.txt'
```
Comment (spytheman) [inline src/uniq/uniq_test.v]: The changes here are unrelated to the purpose of the PR, and over complicate a simple thing. Please revert the file.

File: src/unlink/unlink_test.v [hunk 2]
```diff
@@ -35,19 +35,19 @@ fn test_target_does_not_exist() {
...
 fn test_target_is_directory() {
 	$if !windows {
-		os.mkdir('foo')!
-		assert cmd.same_results('foo')
-		os.rmdir('foo')!
+		os.mkdir('${util}_foo')!
+		assert cmd.same_results('${util}_foo')
+		os.rmdir('${util}_foo')!
 	}
 }
```
Comment (spytheman) [inline src/unlink/unlink_test.v]: same thing, please revert the file

File: src/tac/delete.me
Comment (spytheman) [inline src/tac/delete.me:1]: That file should be removed. Its presence is just to make git checkout the folder, for utils that still have no implementation.

Given that you are implementing tac.v, the folder will be no longer empty, and there is no need for it to have a `delete.me` file.

File: src/tac/settings.autogen.v [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+fn args() Settings {
+	mut fp := app.make_flag_parser(os.args)
+	mut st := Settings{}
+	st.before = fp.bool('before', `b`, false, 'attach the separator before instead of after')
+	st.regex = fp.bool('regex', `r`, false, 'interpret the separator as a regular expression')
+	st.separator = fp.string('separator', `s`, '\n', 'use STRING as the separator instead of newline')
+	mut rem_pars := fp.remaining_parameters()
+	process_args(mut st, mut rem_pars)
+	return st
+}
```
Comment (spytheman) [inline src/tac/settings.autogen.v]: Please merge all .v files here into tac.v .
There is no need to over complicate things, the program is very simple and it will be much easier to be read from top to bottom.

**Comment by JalonSolov** [comment]
He's trying to create a method that would generate the boilerplate (or the majority of it) for all these commands.  That's why the separate files, and some of the other changes.

It would be good to have some consistency across all the commands, since there is a lot of similar boilerplate (all cli commands, all with at least some similarities in options - `-help` at the minimum, etc.).

Not sure if what he is doing is the "best" way to do it, but it is at least an attempt to rein things in.

**Comment by syrmel** [comment]
I can find a way to do that and still avoid file clutter. Will come in a separate PR.

File: src/tac/settings.autogen.v (hunk 3 above)
Comment (syrmel) [reply src/tac/settings.autogen.v]: Done!
